packages/ui/src/CopyInput.tsx

```diff
@@ -21,26 +22,48 @@ export const CopyInput: React.FC<CopyInputProps> = ({
 }) => {
   const { onCopy, hasCopied } = useClipboard(value);
   const theme = useTheme();
+  const buttonWidth = {
+    lg: '115px',
+    md: '100px',
+    sm: '96px',
+  }[size];
+  // Height is input - 2px to account for borders
+  const buttonHeight = theme.components.Input.sizes[size].height - 2;
 
   return (
     <InputGroup width='100%' size={size}>
-      <Input readOnly value={value} width='100%' />
+      <Input
+        readOnly
+        value={value}
+        width='100%'
+        textOverflow='ellipsis'
+        overflow='hidden'
+        paddingRight={buttonWidth}
+      />
       <InputRightElement
         borderLeft={`1px solid ${theme.colors.border.input}`}
-        width={size === 'lg' ? 115 : 100}
-        pr={'1px'}
+        width={buttonWidth}
+        top='1px'
+        right='1px'
+        height='calc(100% - 2px)'
       >
         <Button
           variant='ghost'
-          leftIcon={buttonLeftIcon}
+          leftIcon={
+            <Flex align='center' fontSize='20px'>
+              {buttonLeftIcon}
+            </Flex>
+          }
           onClick={onCopy}
+          borderRadius={theme.components.Input.sizes[size].borderRadius}
           borderTopLeftRadius={0}
           borderBottomLeftRadius={0}
           size={size}
-          height={size == 'lg' ? '46px' : '42px'}
+          height='100%'
```

Comment:
Shouldn't this be `buttonHeight`?
```suggestion
          height={buttonHeight}
```